src/transformers/modeling_utils.py

```diff
@@ -1401,8 +1402,16 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                     model = cls(config, *model_args, **model_kwargs)
         else:
             with no_init_weights(_enable=_fast_init):
+
+                restore_gradient_checkpointing = False
```

Comment:
should probably add some comments and a test here. @sgugger - let me know what if you think the PR would be good for you